jsx/behavioural_feedback_panel.jsx

```diff
@@ -0,0 +1,606 @@
+      {options}
+      </select>
+      </div>
+      </div>
+      </div>
+	}
+
+      var feedback_types = this.props.feedback_types;
+      var input = [];
+      for(var key in feedback_types) {
+        if(feedback_types.hasOwnProperty(key)) {
+          input.push(<option value={feedback_types[key].Type}>{feedback_types[key].Label}</option>)
+    }
+}
+
+      return <div className="panel-body" id ="new_feedback">
+      <div className="form-group">
+      <textarea className="form-control" rows="4" id="comment" value={this.state.text_value} onChange={this.handleTextChange}></textarea>
+      </div>
+	{fieldname_select}
+      <div className="form-group">
+      <div className="row">
+      <label className="col-xs-4">Feedback Type</label>
+      <div className="col-xs-8">
+      <select className="form-control input-sm" name = "input" selected={this.state.input_value} onChange={this.handleInputChange} className="form-control">
+      {input}
+      </select>
+      </div>
+      </div>
+      </div>
+      <div className="form-group">
+      <button id="save_data" onClick={this.createNewThread} className="btn btn-default pull-right btn-sm">Save data</button>
+      </div>
+      </div>
+    }
+  });
+
+var FeedbackSummaryPanel = React.createClass({
+    getInitialState: function(){
+	return{
+	    summary: null
+	}
+    },
+    render: function(){
+	if(this.props.summary_data){
+	    var summary_rows = this.props.summary_data.map(function(row){
+		return <tr>
+		<td>{row.QC_Class}</td>
+		<td><a href={"main.php?test_name=" + row.Instrument + "&candID=" + row.CandID + "&sessionID=" + row.SessionID + "&commentID=" + row.CommentID}>{row.Instrument}</a></td>
+		<td><a href={"main.php?test_name=instrument_list&candID=" + row.CandID + "&sessionID="
+		+ row.SessionID}>{row.Visit}</a></td>
+		<td>{row.No_Threads}</td>
+		</tr>
+	    });
+	}
+
+	if (!(summary_rows === undefined || summary_rows.length == 0)){
+	return <div className="panel-body">
+	<table className = "table table-hover table-primary table-bordered dynamictable">
+	<thead>
+	<tr className="info">
+	<th nowrap="nowrap">QC Class</th>
+
+        <th nowrap="nowrap">Instrument</th>
+
+       <th nowrap="nowrap">Visit</th>
+
+        <th nowrap="nowrap"># Threads</th>
+	</tr>	
+	</thead>
+	<tbody>
+	{summary_rows}
+	</tbody>
+	</table>
+	    </div>
+	}
+	else{
+	    return <div className="panel-body">
+	    This candidate has no behavioural feedback.
+	    </div>
+	}
+    }
+});
+
+   FeedbackPanel = React.createClass({
+    getInitialState: function(){
+      return {
+          threads: '',
+	  summary: null
+      }
+    },
+      componentDidMount: function(){
+	  this.loadSummaryServerData();
+	  
+      var that = this;
+      request = $.ajax({
+        type: "POST",
+        url: "AjaxHelper?Module=bvl_feedback&script=react_get_bvl_threads.php",
```

Comment:
I fixed a bug related to getting the thread (as it was not fetching all the right threads) but I'm not sure if it addresses your issue.